### PR TITLE
Initial work to handle reactive messaging wiring at build time

### LINKED ISSUE #20158: Reactive Messaging should generate ConfigDescriptionBuildItem
### Description

Currently the config editor does not display `mp.messaging` properties. By looking at the annotations in the application we could generate `ConfigDescriptionBuildItem` instances for each incoming/outgoing channel, to allow the user to easily wire them up in the config editor of the Dev UI.

e.g. if you have an `@Incoming("prices")` we should generate a build item for `mp.messaging.incoming.prices.connector`, as well as any other properties that reactive messaging know about. 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by cescoffier**
Initial work to handle reactive messaging wiring at build time.

At the moment it:

0. detects connectors, channels, and orphan channels
1. Auto-associate orphan channel with connectors if there is a single connector
2. Generate configuration description from the connector attributes (#20158)

Fixes #20158

### CODE REVIEW

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringHelper.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        // Avoid direct instantiation
+    }
+
+    /**
+     * Checks if the given set of annotation contains the {@code Connector} qualifier.
+     * 
+     * @param qualifiers the set
+     * @return {@code true} if the set contains {@code Connector}, {@code false} otherwise.
+     */
+    static boolean isConnector(Set<AnnotationInstance> qualifiers) {
```
Comment (ozangunalp) [inline extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringHelper.java:36]: This is unused. I think it won't be used after BeanInfo.getQualifier

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringProcessor.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                } else {
+                    Optional<String> managingConnector = getManagingConnector("incoming", name);
+                    incomings.produce(IncomingChannelBuildItem.of(name, managingConnector.orElse(null)));
+                }
+            }
+        }
+    }
+
+    @BuildStep
+    public void detectOrphanChannels(List<IncomingChannelBuildItem> incomings,
```
Comment (ozangunalp) [inline extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringProcessor.java:105]: I think we need to merge this with the other processor. The `SmallRyeReactiveMessagingProcessor` captured other mediator components except for Connectors and connector managed channels. Now that we have those, we can run the wiring logic (currently running at runtime) at build time, and provide the wiring result to the runtime.

**Review by ozangunalp — CHANGES_REQUESTED** [review]
Needs fixing the rebase, the latest changes to `SmallRyeReactiveMessagingKafkaProcessor` are reverted. And a couple of tests can be nice.

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringProcessor.java (hunk 2 above)
Comment (cescoffier) [reply extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringProcessor.java]: I've merged everything in the `WiringProcessor`.

**Comment by cescoffier** [comment]
Some of these improvements can be seen on the dev UI:

image

**Comment by stuartwdouglas** [comment]
Very nice

**Review by ozangunalp — APPROVED** [review]

**Pull request merged**